src/main/java/bio/terra/service/filedata/flight/ingest/IngestPopulateFileStateFromFileStep.java

```diff
@@ -1,116 +1,70 @@
-public class IngestPopulateFileStateFromFileStep implements Step {
+public abstract class IngestPopulateFileStateFromFileStep implements Step {
   private final LoadService loadService;
   private final int maxBadLines;
   private final int batchSize;
-  private final GcsPdao gcsPdao;
 
   public IngestPopulateFileStateFromFileStep(
-      LoadService loadService, int maxBadLines, int batchSize, GcsPdao gcsPdao) {
+      LoadService loadService, int maxBadLines, int batchSize) {
     this.loadService = loadService;
     this.maxBadLines = maxBadLines;
     this.batchSize = batchSize;
-    this.gcsPdao = gcsPdao;
   }
 
-  @Override
-  public StepResult doStep(FlightContext context) {
-    // Ensure that file ingestion works with extra key-value pairs
+  void readFile(BufferedReader reader, UUID loadId) throws IOException {
     ObjectMapper objectMapper =
         new ObjectMapper()
             .registerModule(new Jdk8Module())
             .registerModule(new JavaTimeModule())
             .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
-
-    FlightMap inputParameters = context.getInputParameters();
-    BulkLoadRequestModel loadRequest =
-        inputParameters.get(JobMapKeys.REQUEST.getKeyName(), BulkLoadRequestModel.class);
-
-    FlightMap workingMap = context.getWorkingMap();
-    UUID loadId = UUID.fromString(workingMap.get(LoadMapKeys.LOAD_ID, String.class));
-    GoogleBucketResource bucketResource =
-        FlightUtils.getContextValue(context, FileMapKeys.BUCKET_INFO, GoogleBucketResource.class);
-    Storage storage = gcsPdao.storageForBucket(bucketResource);
-    String projectId = bucketResource.projectIdForBucket();
     List<String> errorDetails = new ArrayList<>();
-
-    try (BufferedReader reader =
-        new GcsBufferedReader(storage, projectId, loadRequest.getLoadControlFile())) {
-      long lineCount = 0;
-      List<BulkLoadFileModel> fileList = new ArrayList<>();
-
-      for (String line = reader.readLine(); line != null; line = reader.readLine()) {
-        lineCount++;
-
-        try {
-          BulkLoadFileModel loadFile = objectMapper.readValue(line, BulkLoadFileModel.class);
-          fileList.add(loadFile);
-        } catch (IOException ex) {
-          errorDetails.add("Format error at line " + lineCount + ": " + ex.getMessage());
-          if (errorDetails.size() > maxBadLines) {
-            throw new BulkLoadControlFileException(
-                "More than " + maxBadLines + " bad lines in the control file", errorDetails);
-          }
-        }
-
-        // Keep this check and load out of the inner try; it should only catch objectMapper failures
-        if (fileList.size() > batchSize) {
-          loadService.populateFiles(loadId, fileList);
-          fileList.clear();
+    long lineCount = 0;
+    List<BulkLoadFileModel> fileList = new ArrayList<>();
+
+    for (String line = reader.readLine(); line != null; line = reader.readLine()) {
```

Comment:
Another way to write this is
```suggestion
    String line;
    while ((line = reader.readLine()) != null) {
```